### DISCUSSION TITLE
Creating new Repository programmatically with RDF4J-Server/Workbench

### DISCUSSION THREAD

**Discussion by Arc-E-Tect** [eclipse-rdf4j/rdf4j#3130] [Q&A] [answered]
Hi,

I'm looking for an example on how to create a repository programmatically (native or memory/persistent) when using the RDF4J-Server and Workbench running in Tomcat.

I know how to create one using the Workbench, but it would be nice to also have an API that I can just provide with a JSON or something, that would create the repository.

Use-case: I want to be able to spin-up a local configuration based on Docker for testing purposes. I don't need the repository to be available after the tests have run, I just want the repository to be available when my Docker with Tomcat spins up. I do have the RDF4J-Server and Workbench as webapp in the container.

Thanks

**Comment by barthanssens** [comment, marked as answer]
Hi,

the documentation has an example on how to create a repository via the REST API using an RDF file, does that help ?

- https://rdf4j.org/documentation/reference/rest-api/#repository-creation

Best regards

Bart

**Reply by Arc-E-Tect** [reply]
Hi,

That seems to answer my question. Would've been nice to have a CURL example with the example.

```
The given example
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
@prefix rep: <http://www.openrdf.org/config/repository#>.
@prefix sr: <http://www.openrdf.org/config/repository/sail#>.
@prefix sail: <http://www.openrdf.org/config/sail#>.
@prefix ms: <http://www.openrdf.org/config/sail/memory#>.

[] a rep:Repository ;
   rep:repositoryID "test" ;
   rdfs:label "test memory store" ;
   rep:repositoryImpl [
      rep:repositoryType "openrdf:SailRepository" ;
      sr:sailImpl [
	 sail:sailType "openrdf:MemoryStore" ;
	 ms:persist true ;
	 ms:syncDelay 120
      ]
   ].

```
would require the following CURL statement

`curl -X PUT -H "Content-Type: text/turtle" --data-binary @cpm-db.ttl http://localhost:8080/rdf4j-server/repositories/test`

Arc-E-Tect

**Reply by abrokenjester** [reply]
Yeah it's a good idea, more generally I'd like us to switch to using an OpenAPI specification so that we can auto-generate this (e.g. using Swagger) instead of the current hand-written manual.

**Reply by Arc-E-Tect** [reply]
I can recommend using RestDoc by Spring to generate TDD style the documentation and use the OpenAPI extension of RestDoc to generate the OpenAPI spec
